packages/@webex/internal-plugin-dss/test/unit/spec/dss.ts

```diff
@@ -921,6 +1157,79 @@ describe('plugin-dss', () => {
         expect(Batcher.prototype.request.getCall(1).args).to.deep.equal(['id2']);
         expect(result).to.equal(response2);
       });
+
+      it('fails when mercury does not response but only the affected batch', async () => {
+        const requests = [
+          {
+            id: 'req-id-1',
+            resource: '/lookup/orgid/userOrgId/identities',
+            bodyParams: {lookupValues: ['id1', 'id2', 'id3']},
+          },
+        ];
+
+        const {
+          promises: [p1, p2, p3],
+        } = await testMakeBatchedRequests({
+          requests,
+          calls: [
+            {
+              method: 'lookup',
+              params: {id: 'id1', shouldBatch: true},
+            },
+            {
+              method: 'lookup',
+              params: {id: 'id2', shouldBatch: true},
+            },
+            {
+              method: 'lookup',
+              params: {id: 'id3', shouldBatch: true},
+            },
+          ],
+        });
+
+        // only 1 response
+        mercuryCallbacks['event:directory.lookup'](
+          createData('id 1', 0, true, 'lookupResult', {entitiesNotFound: ['id1', 'id13']})
+        );
+
+        // Timeout
+        await clock.tickAsync(6000);
```

Comment:
Not sure exactly what you're trying to test, but if you do this waiting after the second batch, the first batch will still be in the batcher

Reply:
I try to test 2 things 
1. batch fails with timeout
2. batch after the timed out batch finish without any problem.

What do you mean by "the first batch will still be in the batcher"? At the end of the test I am checking the result of the requests and all the request from the first brach are rejected

Reply:
Ok, the description sounded like you wanted to test having two batches in progress at the same time, one of which failed. Maybe change the description to be more clear, like 'fails when mercury does not respond, later batches can still pass ok'. 
You could check that the failure doesn't afffect any other in-progress batches if you wanted, by setting off batch 1, then batch 2, then waiting for batch 1 to timeout, then sending mercury message for batch 2, and chacking only batch 1 promises are rejected. Probably not necessary though.